pygeoapi/linked_data.py

```diff
@@ -173,104 +175,146 @@ def jsonldify_collection(cls, collection, locale_):
+            data['features'][i] = {
+                id_field: identifier,
+                'type': 'schema:Place'
+            }
 
     if data.get('timeStamp', False):
         data['https://schema.org/sdDatePublished'] = data.pop('timeStamp')
 
-    defaultVocabulary = "https://geojson.org/geojson-ld/geojson-context.jsonld"
-    if not geojsonld:
-        ldjsonData = {
-            "@context": [
-                {
-                 "schema": "https://schema.org/",
-                 id_field: "@id",
-                 "type": "@type",
-                },
-                *(context or []),
-                *(geocontext or [])
-            ],
-            **data
-        }
-    else:
-        ldjsonData = {
-            "@context": [
-                defaultVocabulary,
-                *(context or [])
-            ],
-            **data
-        }
+    data['links'] = data.pop('links')
+
+    ldjsonData = {
+        '@context': [defaultVocabulary, *(context or [])],
+        **data
+    }
 
     return ldjsonData
 
 
-def make_jsonld(feature):
-    # Expand properties block
-    feature = {**feature, **feature.get('properties')}
-    feature.pop('properties')
+def jsonldify_geometry(feature):
+    """
+        Render JSON-LD for feature with GeoJSON, Geosparql/WKT, and
+        schema geometry encodings.
+
+        :param feature: feature body to with GeoJSON geometry
+
+        :returns: None
+    """
+
+    geo = feature.get('geometry')
+    geom = asShape(geo)
+
+    # GeoJSON geometry
+    feature['geometry'] = feature.pop('geometry')
+
+    # Geosparql geometry
+    feature['geosparql:hasGeometry'] = {
+        '@type': f'http://www.opengis.net/ont/sf#{geom.geom_type}',
+        'geosparql:asWKT': {
+            '@type': 'http://www.opengis.net/ont/geosparql#wktLiteral',
+            '@value': f'{geom.wkt}'
+        }
+    }
 
-    feature['type'] = 'schema:Place'
+    # Schema geometry
+    feature['schema:geo'] = {'@type': 'schema:GeoShape'}
+    if geom.geom_type == 'Point':
+        feature['schema:geo'] = {
+            '@type': 'schema:GeoCoordinates',
+            'schema:longitude': geom.x,
+            'schema:latitude': geom.y
+        }
+        return
+
+    elif geom.geom_type == 'LineString':
+        _ = [f'{x},{y}' for (x, y) in geom.coords[:]]
+        feature['schema:geo']['schema:line'] = ' '.join(_)
+        return
+
+    elif geom.geom_type == 'MultiLineString':
+        points = list()
+        [points.extend(p.coords[:]) for p in geom.geoms]
+        _ = [f'{x},{y}' for (x, y) in points]
+        feature['schema:geo']['schema:line'] = ' '.join(_)
+        return
+
+    elif geom.geom_type == 'MultiPoint':
+        poly_geom = [(p.x, p.y) for p in geom.geoms]
+        poly_geom.append(poly_geom[0])
+
+    elif geom.geom_type == 'Polygon':
+        poly_geom = geom.exterior.coords[:]
+
+    elif geom.geom_type == 'MultiPolygon':
+        poly = unary_union(geom.buffer(0))
```

Comment:
Perhaps a comment to explain why a geometry is being buffered by zero?

Reply:
Added comment